### ISSUE TITLE
Wrong return value with Responsive Control Number

### ISSUE THREAD

**Issue by steve231293** [elementor/elementor#16032]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description
Hello authors,

I have just detected big bug with Responsive Control Number: This is code to register.
```
$this->add_responsive_control( 'grid_columns', [
	'label'          => __( 'Columns', 'elementor' ),
	'type'           => Controls_Manager::NUMBER,
	'min'            => 1,
	'max'            => 12,
	'step'           => 1,
	'default'        => 3,
	'tablet_default' => 2,
	'mobile_default' => 1,  
] );
```

**Steps to reproduce:** 

1. Add above code to any widgets to test. ( For eg: I added it to: Image Carousel widget )

2. Then create new page and **active all Breakpoints**:
![Screenshot at Aug 18 15-44-46](https://user-images.githubusercontent.com/7134267/129867563-73d782c2-12ee-4003-8774-0a0362488401.png)

3. Drag drop the Widget that you added that code to edit.

4. Input following value:
Widescreen: Grid Columns = 8
Desktop: Grid Columns = 7
Laptop: Grid Columns = 6
Tablet Extra: Grid Columns = 5
Tablet: Grid Columns = 4
Mobile Extra: Grid Columns = 3
Mobile: Grid Columns = 2

5. vardump this variable in function render()
`$this->get_settings_for_display();`

6. Check $settings['grid_columns_mobile_extra']

You can see $settings['grid_columns_mobile_extra'] is empty instead of 3.
![Screenshot at Aug 18 15-57-40](https://user-images.githubusercontent.com/7134267/129869596-6ab8bf26-16d2-4bb0-a92d-7dbd01288b15.png)

![Screenshot at Aug 18 15-53-27](https://user-images.githubusercontent.com/7134267/129868996-23d95b57-e88c-4938-8c10-97b0237ef791.png)

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment
Elementor 3.4.0
Hello Theme.

**Comment by steve231293** [comment]
Please notice that input the correct value that I said ( number deincreaes 8 => 2 )
Because if you input from 7 => 1 it still working normally. I don't have any ideas about this bug.

**Comment by udidol** [comment]
Hi @steve231293,

I followed your instructions to reproduce, using the control you provided above.
I added it into the Heading widget, just to test, since you said it doesn't matter where you add it. 

I got the following output:

![image](https://user-images.githubusercontent.com/36724484/129875120-164d02da-e2f6-4af0-a252-b53e8fe1358f.png)

It looks ok on my end.
Maybe something in the way you added your control inputs is different. Note there are placeholders, make sure the value you see in your input in the Editor is an actual input value and not the placeholder.

**Comment by steve231293** [comment]
Hello @udidol ,

Thanks for quick reply,

Could you confirm you used my code without edit anything?

After some hours to debug. I detect bug caused when use default:

$this->add_responsive_control( 'grid_columns', [
	'label'          => __( 'Columns', 'elementor' ),
	'type'           => Controls_Manager::NUMBER,
	'min'            => 1,
	'max'            => 12,
	'step'           => 1,
	**'default'        => 3,
	'tablet_default' => 2,
	'mobile_default' => 1,**  
] );

**Comment by udidol** [comment]
Yes, I used the code as is.

**Comment by steve231293** [comment]
Hello I try adding code to Heading widget but this bug still showing.

I have just activated Elementor 3.4.0 only.
Hello theme: 2.4.1

Could you re-check to help?

**Comment by steve231293** [comment]
If needed i can share desktop remote for you.

**Comment by steve231293** [comment]
Hello @udidol ,

This bug fixed in Elementor **3.4.1**

Thanks

**Comment by UVLabs** [comment]
I'm still receiving this bug
